proto/src/determined/api/v1/experiment.proto

```diff
@@ -575,3 +575,29 @@ message MoveExperimentRequest {
 
 // Response to MoveExperimentRequest.
 message MoveExperimentResponse {}
+
+// Request to get model definition file tree of an experiment.
+message GetModelDefTreeRequest {
+  // The id of the experiment.
+  int32 experiment_id = 1;
+}
+
+// Response to GetModelDefTreeRequest.
+message GetModelDefTreeResponse {
+  // File tree of an experiment.
+  repeated determined.experiment.v1.FileNode files = 1;
+}
+
+// Request to get a file of model definition.
+message GetModelDefFileRequest {
+  // The id of the experiment.
+  int32 experiment_id = 1;
+  // The path of file.
+  string path = 2;
+}
+
+// Response to GetModelDefFileRequest.
+message GetModelDefFileResponse {
+  // Content of file.
+  bytes file = 1;
```

Comment:
question:  What does the grpc gateway do with bytes objects?  Does it try to treat them like strings?

Also... why bother with a proto version of this REST API when you need the echo version to return the bare file anyway?

Reply:
grpc gateway treats them as json string.

I was under the impression that echo version is only for downloading the file, I'm not sure if it will be the same thing for front end, I will look into it.

Reply:
If it treat it as a json string, this will fail for binary files, because json cannot encode arbitrary binary data.

Reply:
I try to test it with pdf file and the api does not seem to break. Is this because binary data has already been encoded? I imagine that frontend can decode it when needed, but I can definitely make changes if new requests are risen during integration. After all, I don't think we support non-ascii files for viewing. 
<img width="956" alt="Screen Shot 2022-07-05 at 3 26 25 PM" src="https://user-images.githubusercontent.com/40620519/177410440-ec737ef5-fd32-47a6-a9a8-b777765173db.png">

Reply:
Cool, it looks like `bytes`-type objects are getting base64-encoded, which does make it json-safe.

For what it's worth, I verified with a junk file I wrote like this: 
```bash
python -c 'import os; os.write(1, bytes(range(256)))' > junk
```
which is a more reliable way to have invalid utf8.